packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -329,39 +291,43 @@ export const getPropValue = (
  * @param cssOutput CSS and variables to place onto the component
  * @param meta Plugin metadata
  */
-export const buildCompiledComponent = (
-  node: t.JSXElement,
-  cssOutput: CSSOutput,
-  meta: Metadata
-): t.Node => {
-  const { sheets, classNames } = transformCss(cssOutput.css);
+export const buildCompiledComponent = (node: t.JSXElement, cssOutput: CSSOutput): t.Node => {
+  const { classNames } = transformCss(cssOutput.css);
   const classNameProp = node.openingElement.attributes.find((prop): prop is t.JSXAttribute => {
     return t.isJSXAttribute(prop) && prop.name.name === 'className';
   });
+  const joinedClassNames = classNames.join(' ');
 
   if (classNameProp && classNameProp.value) {
     // If there is a class name prop statically defined we want to concatenate it with
     // the class name we're going to put on it.
     const classNameExpression = getPropValue(classNameProp.value);
 
-    const values: t.Expression[] = [t.stringLiteral(classNames.join(' ')) as t.Expression].concat(
-      classNameExpression
-    );
-
     classNameProp.value = t.jsxExpressionContainer(
-      t.callExpression(t.identifier('ax'), [t.arrayExpression(values)])
+      t.templateLiteral(
+        [
+          t.templateElement(
+            {
+              raw: `${joinedClassNames} `,
```

Comment:
Joined classes at build time. Though dynamic values can't be evaluated so sometimes `props.className` is ended up as attribute value.

Reply:
im going to revert this because we still want that behaviour to work :)